pkg/sql/codegen/tensorflow/codegen.go

```diff
@@ -0,0 +1,251 @@
+		cc := fc.(*codegen.CategoryIDColumn)
+		return fmt.Sprintf("tf.feature_column.categorical_column_with_identity(key=\"%s\", num_buckets=%d)",
+			cc.FieldMeta.Name, cc.BucketSize), nil
+	case *codegen.SeqCategoryIDColumn:
+		cc := fc.(*codegen.SeqCategoryIDColumn)
+		return fmt.Sprintf("tf.feature_column.sequence_categorical_column_with_identity(key=\"%s\", num_buckets=%d)",
+			cc.FieldMeta.Name, cc.BucketSize), nil
+	case *codegen.CrossColumn:
+		cc := fc.(*codegen.CrossColumn)
+		var keysGenerated = make([]string, len(cc.Keys))
+		for idx, key := range cc.Keys {
+			if c, ok := key.(codegen.FeatureColumn); ok {
+				code, err := generateFeatureColumnCode(c)
+				if err != nil {
+					return "", err
+				}
+				keysGenerated[idx] = code
+			} else {
+				return "", fmt.Errorf("field in cross column is not a FeatureColumn type: %v", key)
+			}
+		}
+		return fmt.Sprintf(
+			"tf.feature_column.crossed_column([%s], hash_bucket_size=%d)",
+			strings.Join(keysGenerated, ","), cc.HashBucketSize), nil
+	case *codegen.EmbeddingColumn:
+		ec := fc.(*codegen.EmbeddingColumn)
+		catColumn, ok := ec.CategoryColumn.(codegen.FeatureColumn)
+		if !ok {
+			return "", fmt.Errorf("embedding generate code error, input is not featureColumn: %s", ec.CategoryColumn)
+		}
+		sourceCode, err := generateFeatureColumnCode(catColumn)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("tf.feature_column.embedding_column(%s, dimension=%d, combiner=\"%s\")",
+			sourceCode, ec.Dimension, ec.Combiner), nil
+	default:
+		return "", fmt.Errorf("unsupported feature column type %T on %v", c, c)
+	}
+}
+
+func attrToPythonValue(attr interface{}) string {
+	switch attr.(type) {
+	case int:
+		return fmt.Sprintf("%d", attr.(int))
+	case int64:
+		return fmt.Sprintf("%d", attr.(int64))
+	case float32:
+		return fmt.Sprintf("%f", attr.(float32))
+	case float64: // FIXME(typhoonzero): may never use
+		return fmt.Sprintf("%f", attr.(float64))
+	case []int:
+		return intArrayToJSONString(attr.([]int))
+		// TODO(typhoonzero): support []float etc.
+	case string:
+		return attr.(string)
+	default:
+		return ""
+	}
+}
+
+func dtypeToString(dt codegen.FieldType) string {
+	switch dt {
+	case codegen.Float:
+		return "float32"
+	case codegen.Int:
+		return "int64"
+	case codegen.String:
+		return "string"
+	default:
+		return ""
+	}
+}
+
+func isKerasModel(estimator string) (bool, string) {
+	if strings.HasPrefix(estimator, "sqlflow_models.") {
+		return true, estimator
+	}
+	return false, fmt.Sprintf("tf.estimator.%s", estimator)
+}
+
+// Train generates a Python program for train a TensorFlow model.
+func Train(ir codegen.TrainIR) (string, error) {
+	trainParams := make(map[string]interface{})
+	modelParams := make(map[string]interface{})
+	for attrKey, attr := range ir.Attributes {
+		if strings.HasPrefix(attrKey, "train.") {
+			trainParams[strings.Replace(attrKey, "train.", "", 1)] = attr
+		}
+		if strings.HasPrefix(attrKey, "model.") {
```

Comment:
`model.hidden_units = [10, 10]` or `hidden_units = [10, 10]`? Shall we make the model parameters consistent with XGBoost?

Reply:
Should be `model.hidden_units`, to be consistent with all current docs: https://github.com/sql-machine-learning/sqlflow/blob/develop/README.md#quick-overview

Reply:
I see. Let's keep the current way and discuss this in an issue.